scripts/tests/scripts_regression_tests.py

```diff
@@ -1399,6 +1403,29 @@ def test_cime_case(self):
             self.assertEqual(case.get_value("TESTCASE"), "TESTRUNPASS")
 
     ###########################################################################
+    def test_cime_case_prereq(self):
```

Comment:
It seems to me that you should be using the existing system test framework in CIME/SystemTests.
You could create a new test based on the SMS test in which job one is an initial run which writes a restart at the final time and job two is started with the prereq flag and is a CONTINUE_RUN which reads the restart from job 1.  This would test that job 2 does not start before job 1 is complete and that job 2 successfully completes.

Reply:
Sorry for the silence on this; I've been focusing on other work recently.
I'm not certain what the best way to implement a negative test for this is - querying the batch system to verify the job will never be run seems just as much work as checking that the batch system added the dependency, neither of which seem doable with CIME currently.
I'd rather not add a timeout which might not be met even with the prerequisite running successfully, so querying the batch system seems like something I'd have to do.

Reply:
@jedwards4b @jgfouca Would there be any suggestions on a robust method of testing this without querying the queue? Waiting for some threshold and verifying the dependent one didn't run seems dangerous, especially with the recent changes on skybridge. AFAICT, there isn't a robust method, so just doing the simple positive test seems reasonable.

Reply:
The tests I'm envisioning is as follows:
Submit a job (done)
While that job is running, submit a job with the prerequisite argument (the dependent job) for the currently running job (done)
Ensure the job does not run if the other job has not finished successfully yet. (not done)

I can add a check to the dependent job to verify the first job has finished, but if the dependent job was not submitted to the queue correctly due to a bug in the prerequisite logic, this test may still pass if it makes it onto a node after the original finished.
@jedwards4b suggested a negative test, force the original job to fail, and then verify the dependent job is never run. I could add a timeout and verify the job hasn't run by this time; but this depends on the queue not submitting after the timeout.

Another option would be to write a fake batch system as @billsacks suggested; this would be a separate Python script which outputs the arguments to a file so we can verify they're correct. Given all the xml needed to support this fake batch system, I'm not certain this is the way we want to go, but if it's helpful for other parts of the code that could justify it.